### PR TITLE
update variant creator

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

**PR intended to be tested with API branch:** feature/multichannel-multicurrency

### Screenshots
Screenshot 2020-09-24 at 10 51 38
Screenshot 2020-09-24 at 10 52 03
Screenshot 2020-09-24 at 11 00 02

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [ ] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-895 Update variant creator](https://app.clickup.com/t/2549495/SALEOR-895)

File: src/components/PriceField/PriceField.tsx [hunk 1]
```diff
@@ -58,12 +59,20 @@ export const PriceField: React.FC<PriceFieldProps> = props => {
   } = props;
 
   const classes = useStyles(props);
-
+  const hasWrongValue = value < 0;
```
Comment (krzysztofwolski) [inline src/components/PriceField/PriceField.tsx:62]:
```suggestion
  const minValue = 0;
```

File: src/components/PriceField/PriceField.tsx [hunk 2]
```diff
@@ -78,6 +87,10 @@ export const PriceField: React.FC<PriceFieldProps> = props => {
         ),
         type: "number"
       }}
+      inputProps={{
+        min: 0,
```
Comment (krzysztofwolski) [inline src/components/PriceField/PriceField.tsx:91]:
```suggestion
        min: minValue,
```

File: src/components/PriceField/PriceField.tsx [hunk 3]
```diff
@@ -58,12 +59,20 @@ export const PriceField: React.FC<PriceFieldProps> = props => {
...
   return (
     <TextField
       className={className}
-      error={error}
-      helperText={hint}
+      error={error || hasWrongValue}
+      helperText={
+        hint ? (
+          hint
+        ) : hasWrongValue ? (
```
Comment (krzysztofwolski) [inline src/components/PriceField/PriceField.tsx:70]:
```suggestion
        ) : value < minValue ? (
```

File: src/products/components/ProductVariantCreatorPage/ProductVariantCreatorPage.tsx [hunk 4]
```diff
@@ -66,7 +69,13 @@ function canHitNext(
 
       return true;
     case ProductVariantCreatorStep.summary:
-      return data.variants.every(variant => variant.sku !== "");
+      return data.variants.every(
+        variant =>
+          variant.sku !== "" &&
+          variant.channelListings.every(
+            channel => channel.price !== "" && parseInt(channel.price, 10) >= 0
```
Comment (krzysztofwolski) [inline src/products/components/ProductVariantCreatorPage/ProductVariantCreatorPage.tsx:76]: Maybe creating util func for string price check would be good idea?

File: src/products/components/ProductVariantCreatorPage/ProductVariantCreatorPrices.tsx [hunk 5]
```diff
@@ -143,45 +185,61 @@ const ProductVariantCreatorPrices: React.FC<ProductVariantCreatorPricesProps> =
...
+                      </div>
+                      {channelListings?.map(listing => (
+                        <div key={listing.id}>
+                          <Typography
+                            variant="caption"
+                            className={classes.channelName}
+                          >
+                            {listing.name}
+                          </Typography>
+                          <TextField
```
Comment (krzysztofwolski) [inline src/products/components/ProductVariantCreatorPage/ProductVariantCreatorPrices.tsx:208]: Can't we reuse PriceField here?

File: src/products/components/ProductVariantCreatorPage/reducer.ts [hunk 6]
```diff
@@ -94,7 +99,7 @@ function selectValue(
       ? toggle(
           {
             slug: valueSlug,
-            value: ""
+            value: [] // ""
```
Comment (krzysztofwolski) [inline src/products/components/ProductVariantCreatorPage/reducer.ts:102]: Left intentionally?

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: src/products/components/ProductVariantCreatorPage/reducer.ts (hunk 6 above)
Comment (AlicjaSzu) [reply src/products/components/ProductVariantCreatorPage/reducer.ts:102]: i forgot to remove

File: src/products/components/ProductVariantPrice/ProductVariantPrice.tsx [hunk 7]
```diff
@@ -133,16 +132,7 @@ const ProductVariantPrice: React.FC<ProductVariantPriceProps> = props => {
...
-                            error?.length
-                              ? error[0].message
-                              : hasWrongValue
-                              ? intl.formatMessage({
-                                  defaultMessage:
-                                    "Product price cannot be lower than 0."
-                                })
-                              : ""
-                          }
+                          hint={error?.length ? error[0].message : ""}
```
Comment (orzechdev) [inline src/products/components/ProductVariantPrice/ProductVariantPrice.tsx:135]: Where are these errors coming from? Shouldn't they have translations?

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/products/components/ProductVariantCreatorPage/ProductVariantCreatorSummary.tsx [hunk 8]
```diff
@@ -228,32 +243,43 @@ const ProductVariantCreatorSummary: React.FC<ProductVariantCreatorSummaryProps>
...
+                              variantFormErrors.price,
+                              intl
+                            )
+                          : ""
+                      }
+                      value={
+                        variant.channelListings?.find(
+                          channel => channel.channelId === listing.id
+                        )?.price
+                      }
```
[the remark this answers is not in the document]
Comment (AlicjaSzu) [reply src/products/components/ProductVariantCreatorPage/ProductVariantCreatorSummary.tsx:271]: no, it seems to work correctly

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**